### PR TITLE
No need for token to understand HF urls

### PR DESCRIPTION
**Pull request by LysandreJik**
This patches an issue where the token was necessary to understand links to the Hugging Face Hub, which was involuntary behavior.

Fixes https://github.com/huggingface/huggingface_hub/issues/186

### CODE REVIEW

File: src/huggingface_hub/repository.py [hunk 1]
```diff
@@ -265,7 +275,7 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
 
             # checks if repository is initialized in a empty repository or in one with files
             if len(os.listdir(self.local_dir)) == 0:
-                logger.debug(f"Cloning {clean_repo_url} into local empty directory.")
+                logger.warning(f"Cloning {clean_repo_url} into local empty directory.")
```
Comment (LysandreJik) [inline src/huggingface_hub/repository.py:278]: I'm also taking the opportunity to elevate this log's importance, as without it the user might wonder why the Repo instantiation is taking so long. 

Can ignore this in favor of a slight rework of logs as proposed by @thomwolf

File: src/huggingface_hub/repository.py [hunk 2]
```diff
@@ -223,23 +223,30 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
         token = use_auth_token if use_auth_token is not None else self.huggingface_token
         api = HfApi()
 
+        repo_type, namespace, repo_id = repo_type_and_id_from_hf_id(repo_url)
+
+        if namespace is None and token is None:
```
Comment (LysandreJik) [inline src/huggingface_hub/repository.py:228]: This isn't robust to the repositories that live at the root, so I will think a bit more about this.

**Comment by LysandreJik** [comment]
This PR removes the ability to clone from a repository using only its model ID, guessing its namespace in return. This is unfortunately due to the models that live at the root (`bert-base-cased`, `gpt2`).

**Comment by julien-c** [comment]
I didn't read thoroughly but I don't think we care about `Repository` being compatible with top-level (legacy) models, no?

**Comment by LysandreJik** [comment]
Merging with offline approval: this is the most rigorous way to handle things.

**Pull request merged**